src/os/bluestore/BlueStore.cc

```diff
@@ -8083,12 +8083,15 @@ void BlueStore::_fsck_foreach_shared_blob(
 	auto& b = e.blob->get_blob();
 	if (b.is_shared() && passed_sbs.count(e.blob) == 0) {
 	  auto sbid = e.blob->shared_blob->get_sbid();
-	  cb(c->cid, oid, sbid, b);
+	  if (cb(c->cid, oid, sbid, b) == false) {
+	    goto stop_iterating;
+	  }
 	  passed_sbs.emplace(e.blob);
 	}
       } // for ... extent_map
     } // for ... it->valid
   } //if (it(PREFIX_OBJ))
+ stop_iterating:;
```

Comment:
Is there any reason in favor of `goto` than `return` (with a comment maybe)?

Reply:
done